addons/base_vat/models/res_partner.py

```diff
@@ -463,21 +463,26 @@ def check_vat_ar(self, vat):
     def default_compact(self, vat):
         return vat
 
-    def _fix_vat_number(self, vat):
+    def _fix_vat_number(self, vat, country_id):
+        code = self.env['res.country'].browse(country_id).code.lower() if country_id else False
         vat_country, vat_number = self._split_vat(vat)
-        check_func_name = 'compact_' + vat_country
+        if code and code != vat_country:
+            return vat
+        check_func_name = 'compact_' + (code or vat_country)
```

Comment:
Since we know `code` and `vat_country` are equal at this point, it can stay unchanged:
```suggestion
        check_func_name = 'compact_' + vat_country
```